src/content/app/entity-viewer/variant-view/hooks/useExpandedTranscriptConsequence.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect } from 'react';
+import { useAppSelector, useAppDispatch } from 'src/store';
+
+import { getExpandedTranscriptConseqeuenceIds } from '../../state/variant-view/transcriptConsequenceSelectors';
+import { toggleTranscriptConsequenceIds } from '../../state/variant-view/transcriptConsequenceSlice';
+
+type Params = {
+  genomeId: string;
+  variantId: string;
+  alleleId: string;
+  expandIds: string[];
+};
+
+const useExpandedTranscriptConsequence = (params: Params) => {
+  const { genomeId, variantId, alleleId, expandIds } = params;
+  const dispatch = useAppDispatch();
+  const expandedIds = useAppSelector((state) =>
+    getExpandedTranscriptConseqeuenceIds(state, genomeId, variantId, alleleId)
+  );
+
+  useEffect(() => {
+    if (expandedIds === null) {
+      dispatch(
+        toggleTranscriptConsequenceIds(genomeId, variantId, alleleId, expandIds)
+      );
+    }
+  }, []);
```

Comment:
don't we want same check when we switch between genomes, variants, and alleles (i.e. array of dependencies)

Reply:
Oh yeah missed it. But it was working when I tested.